syntax = "proto3";

package music;

service MusicService {
  rpc GetUsers (Empty) returns (UserList);
  rpc GetSongs (Empty) returns (SongList);
  rpc GetPlaylistsByUserId (UserId) returns (PlaylistList);
  rpc GetSongsFromPlaylist (PlaylistId) returns (SongList);
  rpc GetPlaylistsContainingSong (SongId) returns (PlaylistList);
}

message Empty {}

message User {
  int32 id = 1;
  string name = 2;
  int32 age = 3;
}

message Song {
  int32 id = 1;
  string name = 2;
  string artist = 3;
}

message Playlist {
  int32 id = 1;
  string name = 2;
}

message UserList {
  repeated User users = 1;
}

message SongList {
  repeated Song songs = 1;
}

message PlaylistList {
  repeated Playlist playlists = 1;
}

message UserId {
  int32 id = 1;
}

message PlaylistId {
  int32 id = 1;
}

message SongId {
  int32 id = 1;
}
